core/client/src/main/java/alluxio/client/WriteTier.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client;
+
+import alluxio.annotation.PublicApi;
+import alluxio.thrift.TTierPolicy;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Write Tier policy when writing blocks for a file in Alluxio.
+ */
+@PublicApi
+@ThreadSafe
+public enum WriteTier{
+  /**
+   * Write the blocks to the highest tier in the Alluxio Worker.
+   */
+  HIGHEST(1),
+  /**
+   * Preference to write blocks to the highest non-memory tier in an Alluxio Worker.
+   */
+  PREFER_HIGHEST_NON_MEMORY(2),
```

Comment:
Should this not mention memory, since the HIGHEST is not required to be memory?

Also, what does "prefer" imply? Does this mean it will try to write to any tier below the highest, from top to bottom, until it fits?